### PR TITLE
Add/Modify Url Encoding/Decoding for IIS UrlRewrite

### LINKED ISSUE #5985: Add/Modify Url Encoding/Decoding for IIS UrlRewrite
I didn't implement UrlDecode, a server variable in IISUrlRewrite, because there wasn't a great library (or static method) to do it. Right now I'm using Uri.Encode for encoding, which doesn't have the best perf.

I think we can use Uri.EscapeDataString, but fundamentally I think this should be changed to a different library.

### PR DESCRIPTION
**Pull request by senbar**
Add Url encoding and modify url Encoding for IIS UrlRewrite

Add Segment Decode class using Uri.UnescapeDataString and plug it into Input Parser.
Modify Url Encode so that it uses Uri.DecodeDataString.

## Description

As was suggested I have changed Decode Data String to Unescape Data String which is recommended by docs. 
For Url Decoding I have used analogical function  UnescapeDataString. I have written unit tests for decoding covering opposite cases to encoding. I have also written tests for InputParser checking whether decode parameter goes into correct case and correct rule is added.

I wasn't sure if I was supposed to write performance tests since the motivation of changing from UrlEncode was that it's slow. If these test are desired I can add them.

Fixes #5985

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/aspnetcore?pullRequest=40481) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/aspnetcore?pullRequest=40481) before we can accept your contribution.<br/><br/>:x: senbar [sign now](https://cla.dotnetfoundation.org/dotnet/aspnetcore?pullRequest=40481)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/aspnetcore?pullRequest=40481) it.</sub>

File: eng/scripts/InstallVisualStudio.ps1 [hunk 1]
```diff
@@ -35,7 +35,7 @@
 #>
 param(
     [ValidateSet('BuildTools','Community', 'Professional', 'Enterprise')]
-    [string]$Edition = 'Enterprise',
+    [string]$Edition = 'Community',
```
Comment (dougbu) [inline eng/scripts/InstallVisualStudio.ps1:38]: This is a parameter and should not need to be changed. Please revert this file.

**Review by dougbu — COMMENTED** [review]
We don't need the UpgradeLog2.htm file checked in. Likely don't need the many launchSettings.json files either. Suggest removing them from this PR.

File: eng/scripts/InstallVisualStudio.ps1 (hunk 1 above)
Comment (senbar) [reply eng/scripts/InstallVisualStudio.ps1:38]: yes sorry i pushed all untracked files by mistake; I have opened new pull request now with everything in order, sorry about that

**Pull request closed without merging**